### PR TITLE
Solfuzz CPI mocking

### PR DESCRIPTION
**Pull request by ravyu-jump**
For use in CPI fuzzer. Refer to https://github.com/firedancer-io/solfuzz-agave/pull/99/files#diff-890592342949f650be9d1ed182210d7d12be838b2eb2d539a5d0c11f678f1457R169

### CODE REVIEW

File: program-runtime/src/invoke_context.rs [hunk 1]
```diff
@@ -229,6 +231,8 @@ impl<'a> InvokeContext<'a> {
             timings: ExecuteDetailsTimings::default(),
             syscall_context: Vec::new(),
             traces: Vec::new(),
+            #[cfg(feature = "stub-proc-instr")]
+            proc_instr_callback: None,
```
Comment (topointon-jump) [inline program-runtime/src/invoke_context.rs:247]: Do you mind if we change this to be non-optional?

**Review by topointon-jump — APPROVED** [review]

File: program-runtime/src/invoke_context.rs (hunk 1 above)
Comment (ravyu-jump) [reply program-runtime/src/invoke_context.rs:247]: discussed offline: agreed to switch this to non-optional to reduce complexity of fuzz harnesses

Comment (ravyu-jump) [reply program-runtime/src/invoke_context.rs:247]: discussed offline: we are not doing this

**Review by topointon-jump — APPROVED** [review]
LGTM 🔥

File: program-runtime/src/invoke_context.rs [hunk 2]
```diff
@@ -188,6 +188,18 @@ pub struct SerializedAccountMetadata {
...
+// Used to mock InvokeContext::process_instruction 
+#[cfg(feature = "stub-proc-instr")]
+type InstrProcCallback = Box<
+    dyn FnMut(
+        &mut TransactionContext,
+        &[u8], // instruction_data
+        &[InstructionAccount],
+        &[IndexOfAccount] // program_indices
+    ) -> Result<(), InstructionError>
+>;
```
Comment (topointon-jump) [inline program-runtime/src/invoke_context.rs:200]: 🤔

**Pull request merged**